python/cudf/cudf/_lib/utils.pyx

```diff
@@ -138,12 +138,19 @@ cpdef generate_pandas_metadata(Table table, index):
             index_descriptors.append(descr)
 
     metadata = pa.pandas_compat.construct_metadata(
-        table,
-        col_names,
-        index_levels,
-        index_descriptors,
-        index,
-        types,
+        columns_to_convert=[
+            col.to_pandas()
```

Comment:
Why do we need to convert this to pandas now?

Reply:
This was because in latest pyarrow, the `columns_to_convert` is a required parameter and it expects a series. I was passing in column due to which we ran into accessor related errors:

```python
../envs/cudfdev/lib/python3.8/site-packages/cudf/core/dataframe.py:5668: in to_arrow
    metadata = pa.pandas_compat.construct_metadata(
../envs/cudfdev/lib/python3.8/site-packages/pyarrow/pandas_compat.py:213: in construct_metadata
    metadata = get_column_metadata(col, name=sanitized_name,
../envs/cudfdev/lib/python3.8/site-packages/pyarrow/pandas_compat.py:158: in get_column_metadata
    string_dtype, extra_metadata = get_extension_dtype_info(column)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

column = <cudf.core.column.categorical.CategoricalColumn object at 0x7fd0204e4b40>

-- dictionary:
  [
    "A",
    "B",
    "C...   21,
    27,
    ...
    14,
    6,
    43,
    1,
    44,
    28,
    39,
    38,
    13,
    17
  ]
dtype: category

    def get_extension_dtype_info(column):
        dtype = column.dtype
        if str(dtype) == 'category':
            cats = getattr(column, 'cat', column)
            assert cats is not None
            metadata = {
>               'num_categories': len(cats.categories),
                'ordered': cats.ordered,
            }
E           AttributeError: 'function' object has no attribute 'categories'

```

I don't think we would need to convert to pandas now that I have changed this to send `cudf.Series` through this method. I have check internals and they just seems to be extracting metadata such as dtype.